### ISSUE TITLE
Spurious warning/error about calling an abstract method

### ISSUE THREAD

**Issue by johnw42** [google/closure-compiler#2458]
Using the compiler web UI with advanced optimizations turned on, this code produces a spurious warning:

```js
/** @abstract */
class X {
  /** @abstract */
  m1() {}

  m2() {
    return () => this.m1(...[]);
  }
}
```

The warning is:

```
JSC_ABSTRACT_SUPER_METHOD_NOT_CALLABLE: Abstract super method X.prototype.m1 cannot be called at line 9 character 17
    return () => this.m1(...[]);
```

The warning only seems to happen when m1 is called from a lambda using varargs syntax.

**Comment by MatrixFrog** [comment]
This transpiles to

```
X.prototype.m2 = function() {
  /** @const */ var $jscomp$this = this;
  return function() {
    return $jscomp$this.m1.apply($jscomp$this, [].concat($jscomp.arrayFromIterable([])));
  };
};
```

https://jscomp-debug.googleplex.com/#input0%3D%252F**%2520%2540abstract%2520*%252F%250Aclass%2520X%2520%257B%250A%2520%2520%252F**%2520%2540abstract%2520*%252F%250A%2520%2520m1()%2520%257B%257D%250A%250A%2520%2520m2()%2520%257B%250A%2520%2520%2520%2520return%2520()%2520%253D%253E%2520this.m1(...%255B%255D)%253B%250A%2520%2520%257D%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26MISSING_PROPERTIES%3D1%26TRANSPILE%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1

**Comment by MatrixFrog** [comment]
The long-term fix is to have the typechecker handle ES6 code directly. I'm open to suggestions for a shorter term fix as well.

**Comment by MatrixFrog** [comment]
Actually I think I have a fix.
